lib/chef/knife/ec2_base.rb

```diff
@@ -161,6 +161,10 @@ def validate!(keys=[:aws_access_key_id, :aws_secret_access_key])
               Chef::Config[:knife][:aws_access_key_id] = entries['AWSAccessKeyId'] || entries['aws_access_key_id']
               Chef::Config[:knife][:aws_secret_access_key] = entries['AWSSecretKey'] || entries['aws_secret_access_key']
               Chef::Config[:knife][:aws_session_token] = entries['AWSSessionToken'] || entries['aws_session_token']
+              Chef::Config[:knife][:region] = entries['region']
+              if !Chef::Config[:knife][:region]
+                ui.warn "No region was specified in knife.rb or as an argument. The default region, us-east-1, will be used:"
+              end
```

Comment:
Moving the warning here means it only gets printed if the user has an aws credentials file.